src/orquestra/sdk/_base/_driver/_client.py

```diff
@@ -579,15 +590,56 @@ def get_workflow_run_logs(self, wf_run_id: _models.WorkflowRunID) -> bytes:
             ).dict(),
         )
 
+        # Handle errors
         if resp.status_code == codes.NOT_FOUND:
             raise _exceptions.WorkflowRunLogsNotFound(wf_run_id)
         elif resp.status_code == codes.BAD_REQUEST:
             raise _exceptions.InvalidWorkflowRunID(wf_run_id)
 
         _handle_common_errors(resp)
 
-        # TODO: unzip, get logs (ORQSDK-652)
-        return resp.content
+        # Decompress data
+        try:
+            unzipped: bytes = zlib.decompress(resp.content, 16 + zlib.MAX_WBITS)
+        except zlib.error as e:
+            raise _exceptions.WorkflowRunLogsNotReadable(wf_run_id) from e
+
+        untarred = TarFile(fileobj=io.BytesIO(unzipped)).extractfile("step-logs")
+        assert untarred is not None
+        decoded = r"""{}""".format(untarred.read().decode("utf-8"))
+
+        # Parse the decoded data as logs
+        # TODO: index by taskinvocationID rather than workflowrunID [ORQSDK-777]
+        logs = []
+        for section in decoded.split("\n"):
+            if len(section) < 1:
+                continue
+            for log in json.loads(section):
+                try:
+                    # Orquestra logs are jsonable - where we can we parse these and
+                    # extract the useful information
+                    parsed_log_message = json.loads(log[1]["log"])
```

Comment:
Good idea! [0968c89](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/79/commits/0968c890017a6715cc0436fe32cc9d10c8b6be1a) switches to using `WFLog`.

Reply:
Okay, I think rather than implementing another pydantic model we should maybe leave this as-is for now and push ahead with getting a new logs spec agreed with the platform. [ORQSDK-777](https://zapatacomputing.atlassian.net/browse/ORQSDK-777?atlOrigin=eyJpIjoiYjNjZDg2ODk4YTA4NDI3OWEyYzMzODRkZWYyZWM0ZTYiLCJwIjoiaiJ9) is set up to track further efforts.